### DISCUSSION TITLE
Unexpected impact of Python Transform on workflow

### DISCUSSION THREAD

**Discussion by msarvestani** [bonsai-rx/bonsai#1301] [Q&A] [answered]
Hi,
I have a simple workflow that generates flashing circles at different random points on the screen. I log the location, then I feed the x,y location positions to a Python Transform to calculate the vector angle relative to the center of the monitor. This vector will be used later to compare to another.

`import math
from System import Tuple

@returns(Tuple[float,float,float])
def process(value):
  x, y = value
  angle_rad = math.atan2(y, x)
  angle_deg = math.degrees(angle_rad)
  return Tuple.Create(float(x),float(y),float(angle_deg))`

My problem is that if I enable the third float output (Item 3 = vector angle) of the python transform, the location of my DrawCirlce node becomes fixed. If I disable it, all is well.  The reason is Item 1 and Item 2 outputs are empty (not being output) when I enable Item 3. 

I don't understand this behavior at all and wonder if there's a better way to extract the vector angle. 

Many Thanks!

![image](https://user-images.githubusercontent.com/12864873/227238379-1569774e-cea3-4673-8ab7-6d9dad8bc071.png)

**Comment by bruno-f-cruz** [comment]
Could you include the workflow?

**Comment by msarvestani** [comment, marked as answer]
[debug.zip](https://github.com/bonsai-rx/bonsai/files/11056680/debug.zip)
Thanks.

**Reply by bruno-f-cruz** [reply]
This is the expected behavior indeed. Since Item3 is not "bound" by the same "TakeUntil" as the rest of the sequence your "Repeat" node wont be able to resubscribe to the "Range" branch since the `Item3` never closed. 
There are a few ways to fix this, the easiest one is to make sure you connect Item3 back to the main branch. e.g.:
![image](https://user-images.githubusercontent.com/7049351/227466883-1892efe3-f2ca-40a4-b3cd-690a969a361a.png)

**Reply by msarvestani** [reply]
Ok I understand the behavior now and it works. Thanks for the quick reply.
